azure-iot-device/azure/iot/device/provisioning/provisioning_device_client.py

```diff
@@ -77,7 +99,22 @@ def cancel(self):
         logger.info("Cancelling the current registration process")
 
         cancel_complete = EventedCallback()
-        self._polling_machine.cancel(callback=cancel_complete)
+        self._provisioning_pipeline.disconnect(callback=cancel_complete)
         cancel_complete.wait_for_completion()
 
         logger.info("Successfully cancelled the current registration process")
+
+    def _enable_responses(self):
+        """Enable to receive responses from Device Provisioning Service.
+
+        This is a synchronous call, meaning that this function will not return until the feature
+        has been enabled.
+
+        """
+        logger.info("Enabling reception of response from Device Provisioning Service...")
+
+        callback = EventedCallback()
+        self._provisioning_pipeline.enable_responses(callback=callback)
```

Comment:
>enable_responses [](start = 36, length = 16)

this call could raise any of the exceptions that you translate in handle_result above.  You should probably translate those exceptions if they're raised from here too.